src/java/org/apache/cassandra/tools/SSTableExport.java

```diff
@@ -305,6 +313,66 @@ public static void export(Descriptor desc, PrintStream outs, Collection<String>
         }
     }
 
+    public static void export(Descriptor desc, PrintStream outs, String prefix, String[] excludes, CFMetaData metadata) throws IOException
+    {
+        Set<String> excludedKeys = Arrays.stream(excludes).collect(Collectors.toSet());
+        SSTableReader sstable = SSTableReader.open(desc);
+
+        try (RandomAccessReader dfile = sstable.openDataReader())
+        {
+            IPartitioner partitioner = sstable.partitioner;
+
+            outs.println("[");
+
+            // last key to compare order
+            DecoratedKey decoratedKey = partitioner.decorateKey(metadata.getKeyValidator().fromString(prefix));
+            DecoratedKey lastKey = null;
+            int keysCount = 0;
+            while (keysCount < 5000) {
+                if (lastKey != null && lastKey.compareTo(decoratedKey) > 0)
+                    throw new IOException("Key in sstable are out of order despite scanning in order! " + lastKey + " > " + decoratedKey);
+
+                RowIndexEntry entry;
+                if (keysCount == 0) {
+                    entry = sstable.getPosition(decoratedKey, SSTableReader.Operator.GE);
+                } else {
+                    entry = sstable.getPosition(lastKey, SSTableReader.Operator.GT);
+                }
```

Comment:
i think there's two pieces at play here: `sstable.getPosition` will have to binary search for the index for each key but `dfile.seek` will only ever move forward from the previous seeked position. this means searching for a prefix that has 100 keys is the same as if you searched speciically for those 100 keys with `-k`

Reply:
If you had a "next prefix" supplied I think you would be able to run this like a normal range scan right? like "prefix1" is the left bound then "prefix2" is the next?

Reply:
i think with the update this
> this means searching for a prefix that has 100 keys is the same as if you searched speciically for those 100 keys with `-k`
is still the case
```
                RowIndexEntry entry = sstable.getPosition(decoratedKey, keysCount == 0 ? SSTableReader.Operator.GE : SSTableReader.Operator.GT);
```
i think like @zpear said we can just use the on disk atom iterator until we hit untilPrefix (or until the prefix doesn't match anymore) instead of researching for every key